packages/ckeditor5-list/src/todolist/todolistediting.ts

```diff
@@ -392,6 +398,29 @@ export default class TodoListEditing extends Plugin {
 	}
 }
 
+/**
+ * Returns an upcast converter for to-do list items.
+ */
+function todoListItemUpcastConverter(): GetCallback<UpcastElementEvent> {
+	return ( evt, data, conversionApi ) => {
+		const { writer } = conversionApi;
+
+		if ( data.modelRange ) {
+			const items = Array
+				.from( data.modelRange.getItems( { shallow: true } ) )
+				.filter( ( item ): item is Element => item.getAttribute( 'listType' ) === 'todo' );
```

Comment:
This won't work correctly if there is a nested to-do list. It would be handled the same as it would be the same list item. Please remember that lists are stored in the model as a flat structure.

Reply:
@niegowski I added fix to group items before checks. From what I can see it solves the issue.  Can you take a look?